### DISCUSSION TITLE
Updating GrainId after merging grains.

### DISCUSSION THREAD

**Discussion by beroshcr** [mtex-toolbox/mtex#1738] [Ask Anything]
Hi! I am trying to update the parent grainId in my EBSD dataset after accounting for and merging augite grain boundaries. 

There are two augite grain boundaries I've identified and merged reducing my datasets from 102 to 100 defined augite grains. But my code is not successfully updating the grainIds accounting for this merging.

Am I supposed to reconstruct the grains again using this "ebsd_merged" data set? I've tried this and it returns the same 102 grains even after I've "merged" them when reconstructing the grain boundaries. 

How do you successfully combine the two children grains along a twin boundary to produce one mean orientation and incorporate that into a new data set? 

 From the MTEX documentation: 

% copy ebsd data into a new variable to not change the old data
ebsd_merged = ebsd;

% update the grainIds to the parentIds
ebsd_merged('indexed').grainId = parentId(grains.id2ind(ebsd('indexed').grainId))

**Comment by kilir** [comment]
Hi,
1) no, after mergingyour grains, updating your grainId, you should not compute grains again, which obviously will populate the grainId with those resulting from the grain segmentation.
2) twinning in cpx will not produce a reasonable mean orientatio, so it's better not to compute it during merging.
Cheers,
Rüdiger

**Reply by beroshcr** [reply]
Hi Rüdiger,

Thank you for the answer. Perhaps I should add more details, your comments are insightful but I am still having issues. To elaborate further, my goal is to update the grain orientation within the merged child grains to a single crystallographic orientation to produce a single coherent parent grain. I am doing this in order to calculate quantitative metrics to help determine the strength of CPO.

I have successfully created a "mergedGrains" data set that successfully accounts for the merged twin boundaries. I have no problem plotting pole figures with this dataset. However, if I use the mergedGrains in an attempt to calculate my quantitative metrics I receive NaN values (attached). 

The two ending NaN values at the bottom of that initial data list I suspect are due to the two augite grains I've merged. The NaN values are residual from the merging process. 

Now, I am obviously doing something wrong here. Any ideas on how to solve this? Thank you for your help. 


![PGR Values NaN - CB](https://user-images.githubusercontent.com/57849453/220470566-6deee2a0-041d-4431-8d16-09062b3f8d79.png)

**Reply by kilir** [reply]
Hi,
what should be the orientation of the merged grain? One of the two twins? You can compute a mean orientation during merge, or actually use a function handle tellign merge what to do with those orientations, however, a simple mean orientation from twinned grains is usually very wrong. That is why there's no mean orientation computed by default.
Cheers,
Rüdiger

**Reply by beroshcr** [reply]
Hi Rüdiger, 

Yes, I'd like to assign one of the orientations from the children grains to the newly merged parent grain. How do you compute the mean orientation during the merge? What function could I use? 

Cheers,

Christian

**Reply by kilir** [reply]
Hi Christian,

have a look again here: `help merge`. If you use the option 'calcMeanOrientation' if can be followed by a function handle, e.g. https://github.com/mtex-toolbox/mtex/discussions/730#discussioncomment-318754

Cheers,
Rüdiger

**Reply by beroshcr** [reply]
Hi Rüdiger,

That worked! Thank you so much for your help. If you don't mind I would like to ask you one further question. 

I have multiple EBSD data sets for my sample that are not spatially connected (that is they are not adjacent to one another). Is there a way to combine multiple EBSD data sets into one? Or alternatively, is it possible to plot separate EBSD data sets onto the same pole figure? 

Best regards,

Christian

**Comment by lfgmorales** [comment]
Hi

I guess


ebsd=[ebsd1 ebsd2];

Is what you are looking for? This is normally for adjacent maps, but should work to put the databof your maps together

Cheers

Luiz
Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 5 Mar 2023, at 01:15, beroshcr ***@***.***> wrote:

﻿

Hi Rüdiger,

That worked! Thank you so much for your help. If you don't mind I would like to ask you one further question.

I have multiple EBSD data sets for my sample that are not spatially connected (that is they are not adjacent to one another). Is there a way to combine multiple EBSD data sets into one? Or alternatively, is it possible to plot separate EBSD data sets onto the same pole figure?

Best regards,

Christian

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1738#discussioncomment-5204574>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4EY5WQ342AYTR42W3W2PLJNANCNFSM6AAAAAAU6UYOCI>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1738#discussioncomment-5204574", "url": "https://github.com/mtex-toolbox/mtex/discussions/1738#discussioncomment-5204574", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]
